guardian/prosemirror-elements
# Add command to bundle library for distribution

_co-authored-by: @rhystmills_ 

## What does this change?

Adds a new script, `build`, which creates CommonJS and ES6 bundles and Typescript declarations in the `dist` folder, and amends `package.json` to point library consumers to the right files.

This should make the project fit for a 0.1.0 release 🎉 

## How to test

Run `yarn` to ensure deps are up-to-date, and then `yarn build`. You should see files in the `dist` folder corresponding to CommonJS, ES6 and declaration files.

(We also generate declarations for `demo`. This is redundant, but b/c we only refer to the declaration file in `src` in our package, it shouldn't matter, and it'd take a reshuffle of our tsconfigs to sort, so I think we're OK.)

Run `yarn link` in this repository, and then `yarn link prosemirror-elements` in another project in which you're importing it. Attempt to import a function from the library, e.g. 

```import { buildElementPlugin } from 'prosemirror-elements'```

Does your IDE recognise the library typings? Does your consumer build successfully bundle and run the library code?

## How can we measure success?

Our library can be successfully released and imported by consumers.
